hw/ip/i2c/doc/theory_of_operation.md

```diff
@@ -87,23 +87,27 @@ Typically every byte transmitted must also receive an ACK signal, and the IP wil
 However, there are some I2C commands which do not require an ACK.
 In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received.
 
-#### Halt-on-NAK
+#### Halt-on-NACK
 
-If the Host-Mode Controller-Transmitter transmits a byte and the 9th bit is a NACK from the Target/Bus, the `nak` interrupt is usually asserted (modulo the effect of [`FDATA.NAKOK`](registers.md#fdata)).
-If the 'nak' interrupt is asserted, the Byte-Formatted Programming Mode FSM will halt until the interrupt has been acknowledged using the standard 'Event-Type' interrupt acknowledgement (W1C to INTR_STATE).
+If the Host-Mode Controller-Transmitter transmits a byte and the 9th bit is a NACK from the Target/Bus, the `controller_halt` interrupt is usually asserted (modulo the effect of [`FDATA.NAKOK`](registers.md#fdata)).
+If the `controller_halt` interrupt is asserted, the Byte-Formatted Programming Mode FSM will halt until the interrupt condition has been cleared in the [`CONTROLLER_EVENTS.NACK`](registers.md#controller_events) register.
 Software will likely want to do one of two things in handling this irq:
-1. End the current transaction by setting [`CTRL.ENABLEHOST`](registers.md#ctrl) to `1'b0`, and clear the FIFOs ready to begin a new transaction.
-2. Clear the FIFOs, then begin a new transfer without ending the transaction by add new FDATA indicators to the FMTFIFO.
+1. End the current transaction by setting [`CTRL.ENABLEHOST`](registers.md#ctrl) to `1'b0`, and clear the FIFOs to be ready to begin a new transaction.
+2. Clear the FIFOs, then begin a new transfer without ending the transaction by adding new FDATA indicators to the FMTFIFO.
    In this case, the first entry written to the FMTFIFO should have [`FDATA.START`](registers.md#fdata) set to `1'b1` to create a repeated start condition.
 
-While the FSM is halted, SCL/SDA are released and the bus is observed as having an elongated 9th clock period.
+Software should then clear the [`CONTROLLER_EVENTS.NACK`](registers.md#controller_events) to proceed with the next operation.
```

Comment:
I might just add a clarification here that the `controller_halt` interrupt is CIP status-type, and hence does not need to be acknowledged, but will deassert on it's own when software clears the `CONTROLLER_EVENTS` register.